test/AssemblyTests.cmake

```diff
@@ -51,17 +66,50 @@ macro(add_filecheck_test name)
         $<TARGET_OBJECTS:${name}>
         ${ASM_OUTPUT_FILE}
       BYPRODUCTS ${ASM_OUTPUT_FILE})
-  add_dependencies(copy_${name} ${name})
+      add_dependencies(copy_${name} ${name})
+
+  # The main problem here is that the -S flag does not work as clang/gcc in other 
+  # compilers like nvc++ so just add the necessary flags
+  add_library(${name}_normalized OBJECT ${name}.cc)
+  target_link_libraries(${name}_normalized PRIVATE benchmark::benchmark)
+  set_target_properties(${name}_normalized PROPERTIES COMPILE_FLAGS "${ASM_TEST_FLAGS}")
+        
+  # Add a normalized version with objdump + sed script
+  add_custom_target(
+      assembly_${name} ALL
+      COMMAND 
+        objdump -dClG --no-show-raw-insn --section=.text -M att $<TARGET_OBJECTS:${name}_normalized> | 
```

Comment:
do we need to make sure `objdump` is available on the machine running the tests?

Reply:
Great point. I have added `find_program( OBJDUMP objdump )` and respective `if( DEFINED OBJDUMP )` just now.

Reply:
Also added SED